apps/emqx_resource/src/emqx_resource_pool.erl

```diff
@@ -86,10 +86,18 @@ health_check_workers(PoolName, CheckFunc, Timeout, Opts) ->
     Workers = [Worker || {_WorkerName, Worker} <- ecpool:workers(PoolName)],
     DoPerWorker =
         fun(Worker) ->
-            case ecpool_worker:client(Worker) of
-                {ok, Conn} ->
-                    erlang:is_process_alive(Conn) andalso
-                        ecpool_worker:exec(Worker, CheckFunc, Timeout);
+            maybe
+                {ok, Conn} ?= ecpool_worker:client(Worker),
+                true ?= erlang:is_process_alive(Conn),
+                try
+                    ecpool_worker:exec(Worker, CheckFunc, Timeout)
+                catch
+                    exit:{timeout, _} ->
+                        {error, timeout}
```

Comment:
Q: Can the same be achieved by catching `noproc` here? Asking because `is_process_alive/1` involves signalling between processes, and my experience can introduce non-negligible cost in some circumstances.

Reply:
Changed.  Though, this is done during health checks, so probably not such a hot path.

Reply:
Sorry, right, missed that aspect.

Reply:
Oh, wait, it's not quite the same thing: apparently, we check if the inner `Conn` is alive, but call `Worker` if so (different processes).  :thinking: